Datetime

Kawthar1012 opened the pull request.

majafritsch commented on JUnit_tests/DummyTest.java:
```diff
@@ -0,0 +1,11 @@
+        assertEquals(4, new Long(4));
+    }
+
+}
```
Perhaps remove this file since it seems redundant

majafritsch commented on JUnit_tests/test_DateTime.java:
```diff
@@ -0,0 +1,226 @@
+- operations <,>,= to compare dates (check lt,le,eq,... in Int.java)
+- now(), utcnow(), fromtimestamp(timestamp)
+- time()
+ */
```
Maybe remove these comments since they probably don't make sense to someone outside this course. This would be better suited in the pull request description.

majafritsch commented on JUnit_tests/test_DateTime.java:
```diff
@@ -0,0 +1,226 @@
+    }
+
+    @Test
+    public void testTodayFail () {
```
It seems like this test is failing. Perhaps it is intended, but maybe there is a way to expect failure somehow, so that the test passes

majafritsch commented on JUnit_tests/test_DateTime.java:
```diff
@@ -0,0 +1,226 @@
+    }
+
+    @Test
+    public void testToOrdinal () {
```
This test also seems to fail.

majafritsch commented on python/common/org/python/stdlib/datetime/DateTime.java:
```diff
@@ -0,0 +1,392 @@
+        return String.valueOf((year-1)*dayInNormalYear + checkDaysBeforeMonth(month,year) + day + checkHowManyLeapYear(year));
+    }
+
+    public static org.python.types.Bool __lt__ (DateTime date, DateTime date2) {
```
We noticed that you declared the comparison methods as static. We did not declare ours as static, but we're not sure which way is the correct way. Maybe we could discuss this at the sprint review and our next meeting.

majafritsch commented on JUnit_tests/test_DateTime.java:
```diff
@@ -0,0 +1,226 @@
+    public void testInstanceError1 () {
+        Assertions.assertThrows(SyntaxError.class, () -> {
+            org.python.types.Int[] args = dateTimeArgs(2021,9,15,10,00,01,2759);
+            //java.util.Map<java.lang.String, org.python.Object> kwargs = dateTimeKwargs(2021,9,16,10,00,01,2759);
```
Remove comment

majafritsch commented on JUnit_tests/test_DateTime.java:
```diff
@@ -0,0 +1,226 @@
+    public void testFromOrdinal () {
+        DateTime test = (DateTime) DateTime.today();
+        long ordinal = 738052;
+//        Equivalent to 2021-09-19 -> 738052
```
Remove comment

danielsson-maja reviewed (commented):
Great job!! We have now reviewed your pull request and have some comments that you could look in to.

- Remove .jar and .class-files from pull request
- Maybe remove the Date.java and TimeDelta.java from the pull request if you haven't made changes to them and don't need them for your own module.
- We are a bit unsure about the _python/common/python/datetime.java_  file since it is not located in _python/common/org/python/stdlib/datetime/ _. Is it supposed to be included in the pull request?
- Review the indentations of all the functions. It seems like they are not indented with four whitespaces.
- In some function declarations, if statements and for loops there seems to be a whitespace missing next to the parentheses, e.g. on row 245 and 250 in _python/common/org/python/stdlib/datetime/DateTime.java_   (_ our convention says: _if (delta == null) {_
- Some methods have CamelCased names, but our conventions states that methods should be written in snake_case.
- We think it would be nice with a few more test cases for each of methods to ensure that they work correctly. It seems like they do (:D), but when we looked at the code we felt the urge to add more tests with some edge cases just because we weren't sure they were handled. 

Apart from these things we think you've done a great job! :)

Kawthar1012 commented:
Thank you for your review! We deleted everything that was unnecessary and added test as suggested. According to our IDE, the indentations have indeed four white spaces so it might be a difference depending on which one you use to see the files.

Kawthar1012 commented:
Got what you meant by the indentations, it is now fixed. Thank you!
